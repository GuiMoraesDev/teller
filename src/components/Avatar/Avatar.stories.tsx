import React from 'react'

import { ComponentMeta, ComponentStory } from '@storybook/react'

import Avatar from '.'

const Storie = {
  title: 'Components/Avatar',
  component: Avatar
} as ComponentMeta<typeof Avatar>

export default Storie

const Template: ComponentStory<typeof Avatar> = (args) => <Avatar {...args} />

export const Default = Template.bind({})

Default.args = {
  src: 'https://i.pravatar.cc/150?id=50',
	alt: 'avatar placeholder'
}

export const Bordered = Template.bind({})

Bordered.args = {
  src: 'https://i.pravatar.cc/150?id=50',
	alt: 'avatar placeholder',
	isBordered: true,
}
