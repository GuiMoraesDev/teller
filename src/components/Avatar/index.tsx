import * as Styles from './styles';

export interface StyledProps {
	isBordered?: boolean;
}

interface Props extends StyledProps {
	src: string;
	alt: string;
}

const Avatar = ({ src, alt, isBordered = false }: Props): JSX.Element => {
	return (
		<Styles.AvatarWrapper>
			<Styles.Avatar
				src={src}
				alt={alt}
				isBordered={isBordered}
				referrerPolicy="no-referrer"
			/>
		</Styles.AvatarWrapper>
	);
};

export default Avatar;
