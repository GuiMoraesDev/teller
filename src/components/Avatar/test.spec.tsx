import React from 'react';

import { render } from 'tests/test.utils';

import Avatar from './index';

describe('Avatar component', () => {
	it('should render correctly without isBordered prop', () => {
		const { container } = render(
			<Avatar src="https://i.pravatar.cc/150?id=50" alt="avatar placeholder" />
		);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render correctly with isBordered prop', () => {
		const { container } = render(
			<Avatar
				src="https://i.pravatar.cc/150?id=50"
				alt="avatar placeholder"
				isBordered
			/>
		);

		expect(container.firstChild).toMatchSnapshot();
	});
});
