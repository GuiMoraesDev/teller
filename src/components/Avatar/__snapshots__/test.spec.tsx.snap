// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render correctly with isBordered prop 1`] = `
.c0 {
  height: 100%;
  border: none;
}

.c1 {
  max-width: 100%;
  max-height: 100%;
  border-radius: 10000px;
  border: 3px solid #FDFDFD;
}

<span
  class="c0"
>
  <img
    alt="avatar placeholder"
    class="c1"
    src="https://i.pravatar.cc/150"
  />
</span>
`;

exports[`Avatar component should render correctly without isBordered prop 1`] = `
.c0 {
  height: 100%;
  border: none;
}

.c1 {
  max-width: 100%;
  max-height: 100%;
  border-radius: 10000px;
  border: none;
}

<span
  class="c0"
>
  <img
    alt="avatar placeholder"
    class="c1"
    src="https://i.pravatar.cc/150"
  />
</span>
`;
